pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -1129,8 +1119,34 @@ protected IdealState validateLLCSegments(final TableConfig tableConfig, IdealSta
               committingSegmentDescriptor);
 
           consumingSegments.add(newLLCSegmentName.getSegmentName());
+        } else if (isAllInstancesInState(instanceStateMap,
+            PinotHelixSegmentOnlineOfflineStateModelGenerator.ONLINE_STATE)) { // else the segment is in ONLINE state
```

Comment:
checking for all states ONLINE will not work all the time (what if one of the realtime hosts is OFFLINE)?

We should be checking for absence of CONSUMING segments

Reply:
Combined the else condition to handle absence of CONSUMING in general. The steps to repair are same no matter what we are left with in the absence of CONSUMING segments